### PR TITLE
Game have states, an history and maps

### PR DESCRIPTION
**Pull request by lgilliand**

### CODE REVIEW

**Review by Laykel — CHANGES_REQUESTED** [review]
You knew what I was going to say, right?

We have a serious case of missing Javadoc, here. 
We need to keep a certain level of quality, even if we're very late. ^^

**Comment by Laykel** [comment]
Apart from that, it looks good to me. It felt weird at first to have the key listeners in a class called `MapState`, but I think it's alright.

File: src/main/java/ch/heigvd/mcr/bridgehack/MapState.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    private boolean turnIsOver = false;
+    private int turn = 0;
+    private int counter = 0;
+
+    @Override
+    public void init(GameContainer gameContainer, StateBasedGame stateBasedGame) throws SlickException {
+        maps = new LinkedList<>();
+        maps.add(new Map(1, false));
+        maps.add(new Map(2, true));
+        map = maps.get(0);
```
Comment (Tatallias) [inline src/main/java/ch/heigvd/mcr/bridgehack/MapState.java]: 3rd map is missing

**Comment by Tatallias** [comment]
Besides the changes already called out, this PR seems good.

File: src/main/java/ch/heigvd/mcr/bridgehack/MapState.java (hunk 1 above)
Comment (Laykel) [reply src/main/java/ch/heigvd/mcr/bridgehack/MapState.java]: But don't add it, since @Tatallias has done it in #31.

**Review by Laykel — APPROVED** [review]

**Pull request merged**